app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt

```diff
@@ -524,12 +536,72 @@ private fun InstructionView(currentRecipe: Recipe) {
                   top = OVERVIEW_INSTRUCTION_TOP.dp,
                   bottom = OVERVIEW_INSTRUCTION_BOTTOM.dp)
               .testTag(INSTRUCTIONS_VIEW)) {
-        // Display of the instructions
+        for (index in currentRecipe.instructions.indices) {
+          InstructionValue(currentRecipe.instructions[index], index)
+        }
+      }
+}
 
-        Text(
-            text = currentRecipe.instructions,
-            color = Color.Black,
-            style = MaterialTheme.typography.bodyMedium,
-            modifier = Modifier.testTag(INSTRUCTIONS_TEXT))
+@Composable
+fun InstructionValue(instruction: Instruction, index: Int) {
+  val officialStep = index + 1
+  var isExpanded by remember { mutableStateOf(false) } // Track the expanded state
+
+  Card(
+      modifier =
+          Modifier.fillMaxWidth(1f)
+              .padding(horizontal = MEDIUM_PADDING.dp, vertical = REALLY_SMALL_PADDING.dp)
+              .clickable(onClick = { isExpanded = !isExpanded })
+              .shadow(
+                  elevation = CARD_SHADOW_ELEVATION.dp,
+                  shape = RoundedCornerShape(CARD_BORDER_ROUND.dp))
+              .testTag("InstructionValue"),
+      colors =
+          CardDefaults.cardColors(
+              containerColor = MaterialTheme.colorScheme.background,
+              contentColor = MaterialTheme.colorScheme.onPrimary),
+      shape = RoundedCornerShape(CARD_BORDER_ROUND.dp)) {
+        Row(
+            modifier = Modifier.fillMaxWidth(ROW_SIZE).padding(SMALL_PADDING.dp),
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically) {
+              Image(
+                  painter = painterResource(id = instruction.icon.iconResId),
+                  contentDescription = "Icon",
+                  modifier = Modifier.size(ICON_SIZE.dp).testTag("InstructionIcon"))
+
+              Column(modifier = Modifier.testTag("InstructionInfo")) {
+                Text(
+                    modifier = Modifier.testTag("InstructionTitle"),
+                    color = Color.Black,
```

Comment:
**Hardcoded Colors:**
   - Avoid hardcoding colors like `Color.Black`. Use `MaterialTheme.colorScheme` instead to support light/dark themes dynamically.
   
   **Replace:**
   ```kotlin
   color = MaterialTheme.colorScheme.onPrimary
   ```